Tests/src/PublicAPITests/AssetSystemTests.cpp

```diff
@@ -2168,4 +2162,156 @@ CSP_PUBLIC_TEST(CSPEngine, AssetSystemTests, DownloadAssetDataInvalidURLTest)
+{
+	SetRandSeed();
+
+	auto& SystemsManager = csp::systems::SystemsManager::Get();
+	auto* UserSystem	 = SystemsManager.GetUserSystem();
+	auto* AssetSystem	 = SystemsManager.GetAssetSystem();
+	auto* SpaceSystem	 = SystemsManager.GetSpaceSystem();
+
+	const char* TestSpaceName			= "OLY-UNITTEST-SPACE-REWIND";
+	const char* SpaceDescription		= "OLY-UNITTEST-SPACEDESC-REWIND";
+	const char* TestAssetCollectionName = "OLY-UNITTEST-ASSETCOLLECTION-REWIND";
+	const char* TestAssetName			= "OLY-UNITTEST-ASSET-REWIND";
+	auto FilePath						= std::filesystem::absolute("assets/test.json");
+
+	char SourceSpaceName[256];
+	SPRINTF(SourceSpaceName, "%s-%s", TestSpaceName, GetUniqueString().c_str());
+
+	char DestSpaceName[256];
+	SPRINTF(DestSpaceName, "%s-%s", TestSpaceName, GetUniqueString().c_str());
+
+	csp::systems::AssetCollection SourceAssetCollection;
+
+	// Log in
+	csp::common::String UserId;
+	LogIn(UserSystem, UserId);
+
+	// Create 'source' space and asset collection
+	csp::systems::Space SourceSpace;
+	{
+		printf("Creating source space and asset collection.\n");
+
+		CreateSpace(SpaceSystem, SourceSpaceName, SpaceDescription, csp::systems::SpaceAttributes::Private, nullptr, nullptr, nullptr, SourceSpace);
+
+		char AssetCollectionName[256];
+		SPRINTF(AssetCollectionName, "%s-%s", TestAssetCollectionName, GetUniqueString().c_str());
+
+		char AssetName[256];
+		SPRINTF(AssetName, "%s-%s", TestAssetName, GetUniqueString().c_str());
+
+		// Create an asset collection that belongs to the source space with a single valid asset
+		CreateAssetCollection(AssetSystem, SourceSpace.Id, nullptr, AssetCollectionName, nullptr, nullptr, SourceAssetCollection);
+
+		// Create an asset that belongs to the source collection
+		csp::systems::Asset Asset;
+		CreateAsset(AssetSystem, SourceAssetCollection, AssetName, nullptr, nullptr, Asset);
+
+		// Upload data for the source asset we have created
+		csp::systems::FileAssetDataSource Source;
+		Source.FilePath = FilePath.u8string().c_str();
+		Source.SetMimeType("application/json");
+
+		printf("Uploading source asset data...\n");
+
+		csp::common::String Uri;
+		UploadAssetData(AssetSystem, SourceAssetCollection, Asset, Source, Uri);
+	}
+
+	// Create 'dest' space and invoke the copy
+	csp::systems::Space DestSpace;
+	csp::common::Array<csp::systems::AssetCollection> DestAssetCollections;
+	{
+		printf("Creating dest space and invoking the copy...\n");
+
+		CreateSpace(SpaceSystem, DestSpaceName, SpaceDescription, csp::systems::SpaceAttributes::Private, nullptr, nullptr, nullptr, DestSpace);
+
+		csp::common::Array<csp::systems::AssetCollection> SourceAssetCollections(&SourceAssetCollection, 1);
+		auto [Result] = AWAIT_PRE(AssetSystem, CopyAssetCollectionsToSpace, RequestPredicate, SourceAssetCollections, DestSpace.Id, false);
+
+		EXPECT_EQ(Result.GetResultCode(), csp::systems::EResultCode::Success);
+		DestAssetCollections = Result.GetAssetCollections();
```

Comment:
Surround test macros with blank lines to separate them from other parts of the test.

Reply:
Addressed in latest